### PR TITLE
added change user password route

### PR DESCRIPTION
**Pull request by Carl-Elkhoury**
## Description

Added a new route to change password of user

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/controllers/userAuth.ts [hunk 1]
```diff
@@ -197,6 +197,50 @@ const userAuthController = {
...
+            res.status(statusCodes.MISSING_PARAMS).json(
+                {
+                    status: 422,
+                    message: `Error: there are missing parameters.`
+                }
+            );
+        } else {
+            const oldPassword = req.body.oldPassword;
+            const newPassword = req.body.newPassword;
+            const jwtToken = req.body.jwtToken;
```
Comment (preyansh98) [inline src/controllers/userAuth.ts:215]: can we change from req.body.token to 

`request.headers?.authorization.split(" ")[1]` 

i.e. get token from request headers just to keep it consistent with how we're getting it for other api's

Comment (preyansh98) [inline src/controllers/userAuth.ts]: can we change from req.body.token to

`request.headers?.authorization.split(" ")[1]
`
i.e. get token from request headers just to keep it consistent with how we're getting it for other api's

File: src/controllers/userAuth.ts [hunk 2]
```diff
@@ -197,6 +197,50 @@ const userAuthController = {
...
+                        }
+                    );
+                } else {
+                    let userData : IUserModel;
+                    const userId = decoded._id;
+                    try {
+                        userData = await userDBInteractions.findWithPassword(userId);
+                        const result = await compareHash(oldPassword, userData.password);;
+                        if (!result) {
+                            throw new Error('Wrong old password');
```
Comment (preyansh98) [inline src/controllers/userAuth.ts]: this wont print in .json(error)
either `throw { "error" : "Wrong old password" `} or  `                    res.status(statusCodes.BAD_REQUEST).json(
                        {
                            status: statusCodes.UNAUTHORIZED,
                            message: 'Wrong old password'
                        }
                    );`

File: src/validators/auth.ts [hunk 3]
```diff
@@ -23,5 +23,12 @@ export function authValidator(method: string): ValidationChain[] {
                 body("password", "Invalid or missing 'password'").isString().exists(),
             ];
         }
+        case "POST /passwordReset": {
+            return [
+                body("oldPassword", "Invalid or missing 'oldPassword'").isString().exists(),
+                body("newPassword", "Invalid or missing 'newPassword'").isString().exists(),
+                body("jwtToken", "Invalid or missing 'jwtToken'").isString().exists()
```
Comment (preyansh98) [inline src/validators/auth.ts]: check for auth header after moving token out of body

**Review by preyansh98 — COMMENTED** [review]
LGTM apart from the nits

File: src/validators/auth.ts (hunk 3 above)
Comment (preyansh98) [reply src/validators/auth.ts]: something like header("authorization", "invalid or missing token").exists() might work? not sure

File: src/controllers/userAuth.ts (hunk 2 above)
Comment (Carl-Elkhoury) [reply src/controllers/userAuth.ts]: Sounds good will change that

**Review by preyansh98 — APPROVED** [review]

**Pull request merged**